### ISSUE TITLE
data

### ISSUE THREAD

**Issue by chenjunxia06** [Transconnectome/SwiFT#2]
Hi, my training data shape means [fmri.shape]=[(2296,64,64,30,1)]. So while processing the data, do I similarly populate the data with 64,64,30 to 96*96*96? Is there not an error in my comprehension here? I will be grateful if you can resolve my doubts.

**Comment by Padraig20** [comment]
**(not an author, just someone who is interested)**

Hi @chenjunxia06,
I hope I understand your question correctly. You have an input size of *[64,64,30,1]* and are asking if you have to transform it to *[96,96,96,1]*?
In that case, I strongly believe that you do not have to do this. In the paper, the authors mentioned...

> To easily divide the volume into patches for SwiFT, we changed the 3D volume into a shape of 96 × 96 × 96 by cropping and padding on the background.

Therefore I believe that it is recommended for easily dividing the volume into patches, but not entirely necessary. You do, however, have to find a suitable patch size for your data, which I personally failed to do. Essentially, you need to make sure that the dimensions are divisible by 2 in upcoming stages after dividing it by the patch size.

Here's an example for input size *[2296,64,64,64,1]* and patch size *[4,4,4,1]*:

-   **Initial Input**: (2296,64,64,64,1)
-   **After Stage 1**: (2296,16,16,16,216)
	- initial input has been divided by patch size (4)
-   **After Stage 2**: (2296,8,8,8,432)
	- output of stage 2 is divided by 2
-   **After Stage 3**: (2296,4,4,4,864)
	- output of stage 3 is divided by 2
-   **After Stage 4**: (2296,2,2,2,1728)
	- output of stage 3 is divided by 2

I experimented around with the Dummy dataset and the given input size and patch size, you can find a screenshot of how data passes through the architecture here:

![image](https://github.com/Transconnectome/SwiFT/assets/111874815/76696c7b-4ff8-4eda-aeb4-207d42622d9c)

For your case, I feel like dimension 30 is an issue. If you use an incompatible patch size (i.e. the output of further stages is not divisible by 2), this will result in rounding errors and thus incompatible tensor dimensions. For *D=30* valid divisors are 1,2,3,5,6,10,15,30, but none of these would result in a dimension that is a power of two, further complicating the situation.

![image](https://github.com/Transconnectome/SwiFT/assets/111874815/8a82319e-e951-4784-a8b0-61c383658007)

I suppose that changing the input size to e.g. *(64, 64, 32)* may help you with your situation. Upsampling to *(96, 96, 96)* might just add too much noise and have a bad effect on the model's performance.

Might be wrong, but I hope this helps! :)

**Comment by chenjunxia06** [comment]
Ok, your reply is very helpful, thanks for answering!
